content/en/docs/overview/interfaces.md

```diff
@@ -3,7 +3,141 @@ title: "Interfaces"
 weight: 280
 ---
 
-## Goals
 
+## Basics
+
+Interfaces are used to express general behaviour across multiple types.
+Interface types are defined as a set of method signatures.
+All types which implement this set of methods implement the defined interface.
+
+As an example we define the interface `Stringer`:
+```golang
+type Stringer interface {
+	String() string
+}
 ```
+
+All type which implement the method `String()` which returns a `string` implement the `Stringer` interface.
+Types implicitly implement an interface if they implement the required methods. Other than in for example Java we don't have to explicitly specify that a certain type implements an inteface (`implements`).
+
+
+In the following example `User` and `Group` implement the `Stringer` interface.
+The `PrintAll` function expects a slice of `Stringer`s. This way we can pass users, groups and every other type which implements the `Stringer` interface to the `PrintAll` function.
+In the main function we then initialize a user and a group and put them both into a slice of `Stringer`s which we then pass to the `PrintAll` function.
+
+
+{{<go-playground>}}
+package main
+
+import "fmt"
+
+// interface type Stringer
+type Stringer interface {
+	String() string
+}
+
+// concrete type User
+type User struct {
+	Username string
+}
+
+func (u *User) String() string {
+	return u.Username
+}
+
+// concrete type Group
+type Group struct {
+	Groupname string
+}
+
+func (g *Group) String() string {
+	return g.Groupname
+}
+
+// a function which takes a list of Stringers
+func PrintAll(items []Stringer) {
+	for _, item := range items {
+		// the only thing we know about the items is that they have a method String()
+		text := item.String()
+		fmt.Println(text)
+	}
+}
+
+func main() {
+	user1 := &User{
+		Username: "andrea",
+	}
+
+	group1 := &Group{
+		Groupname: "admins",
+	}
+
+	myItems := []Stringer{user1, group1}
+
+	PrintAll(myItems)
+}
+<!--output-->
+andrea
+admins
+{{</go-playground>}}
+
+
+## Empty Interface
```

Comment:
I think it is important that you know what they are, because they are used in couple of widely used standard library packages:
* JSON and other encoding packages. e.g.: https://pkg.go.dev/encoding/json#Marshal
* All print functions in fmt. e.g.: https://pkg.go.dev/fmt#Println
* In crypto private and public keys are also expressed as empty interface https://pkg.go.dev/crypto#PublicKey